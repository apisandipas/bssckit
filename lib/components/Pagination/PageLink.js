"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _theme = _interopRequireDefault(require("../../theme"));

var _themeFunctions = require("../../utils/themeFunctions");

var _base = require("../Utilities/base");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  display: block;\n  color: ", " !important;\n  cursor: pointer;\n  text-decoration: none;\n  background-color: ", ";\n  border: ", "\n    ", ";\n\n  &:hover {\n    z-index: 2;\n    color: ", ";\n    text-decoration: underline;\n    background-color: ", ";\n  }\n\n  &:focus {\n    z-index: 3;\n    outline: 0;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var PageLink = (0, _styledComponents["default"])(function (_ref) {
  var Component = _ref.as,
      props = _objectWithoutProperties(_ref, ["as"]);

  return _react["default"].createElement(Component, props);
})(_templateObject(), function (props) {
  return (0, _themeFunctions.getColor)(props, 'paginationLink', 'color');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'paginationLink', 'backgroundColor');
}, function (props) {
  return (0, _themeFunctions.getBorder)(props, 'paginationLink', 'default');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'paginationLink', 'borderColor');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'paginationLink', 'hoverColor');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'paginationLink', 'hoverBackground');
});
PageLink.defaultProps = {
  theme: _theme["default"],
  as: _base.A
};
var _default = PageLink;
exports["default"] = _default;