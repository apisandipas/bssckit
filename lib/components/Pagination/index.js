"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Pagination", {
  enumerable: true,
  get: function get() {
    return _Pagination["default"];
  }
});
Object.defineProperty(exports, "PageLink", {
  enumerable: true,
  get: function get() {
    return _PageLink["default"];
  }
});
Object.defineProperty(exports, "PageItem", {
  enumerable: true,
  get: function get() {
    return _PageItem["default"];
  }
});

var _Pagination = _interopRequireDefault(require("./Pagination"));

var _PageLink = _interopRequireDefault(require("./PageLink"));

var _PageItem = _interopRequireDefault(require("./PageItem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }