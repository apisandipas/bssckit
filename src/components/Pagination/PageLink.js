import React from 'react';
import styled, { css } from 'styled-components';
import defaultTheme from '../../theme';
import { getColor, getBorder } from '../../utils/themeFunctions';
import { A } from '../Utilities/base';

const PageLink = styled(({ as: Component, ...props }) => <Component {...props} />)`
  position: relative;
  display: block;
  color: ${props => getColor(props, 'paginationLink', 'color')} !important;
  cursor: pointer;
  text-decoration: none;
  background-color: ${props => getColor(props, 'paginationLink', 'backgroundColor')};
  border: ${props => getBorder(props, 'paginationLink', 'default')}
    ${props => getColor(props, 'paginationLink', 'borderColor')};

  &:hover {
    z-index: 2;
    color: ${props => getColor(props, 'paginationLink', 'hoverColor')};
    text-decoration: underline;
    background-color: ${props => getColor(props, 'paginationLink', 'hoverBackground')};
  }

  &:focus {
    z-index: 3;
    outline: 0;
  }
`;

PageLink.defaultProps = {
  theme: defaultTheme,
  as: A,
};

export default PageLink;
