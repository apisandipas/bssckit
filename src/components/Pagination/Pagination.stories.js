import React from 'react';
import { Pagination, PageItem, PageLink } from './index';

export default {
  title: 'Pagination',
};

export const simple = () => (
  <nav aria-label='Page navigation'>
    <Pagination>
      <PageItem>
        <PageLink to='#'>
          <span>&laquo;</span>
        </PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>1</PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>2</PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>3</PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>
          <span>&raquo;</span>
        </PageLink>
      </PageItem>
    </Pagination>
  </nav>
);

export const activeAndDisabledStates = () => (
  <nav aria-label='Page navigation'>
    <Pagination>
      <PageItem disabled>
        <PageLink to='#'>
          <span>&laquo;</span>
        </PageLink>
      </PageItem>

      <PageItem active>
        <PageLink to='#'>1</PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>2</PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>3</PageLink>
      </PageItem>

      <PageItem>
        <PageLink to='#'>
          <span>&raquo;</span>
        </PageLink>
      </PageItem>
    </Pagination>
  </nav>
);

export const sizes = () => (
  <>
    <nav aria-label='Page navigation'>
      <Pagination lg>
        <PageItem disabled>
          <PageLink to='#'>
            <span>&laquo;</span>
          </PageLink>
        </PageItem>

        <PageItem active>
          <PageLink to='#'>1</PageLink>
        </PageItem>

        <PageItem>
          <PageLink to='#'>2</PageLink>
        </PageItem>

        <PageItem>
          <PageLink to='#'>3</PageLink>
        </PageItem>

        <PageItem>
          <PageLink to='#'>
            <span>&raquo;</span>
          </PageLink>
        </PageItem>
      </Pagination>

      <nav aria-label='Page navigation'>
        <Pagination sm>
          <PageItem disabled>
            <PageLink to='#'>
              <span>&laquo;</span>
            </PageLink>
          </PageItem>

          <PageItem active>
            <PageLink to='#'>1</PageLink>
          </PageItem>

          <PageItem>
            <PageLink to='#'>2</PageLink>
          </PageItem>

          <PageItem>
            <PageLink to='#'>3</PageLink>
          </PageItem>

          <PageItem>
            <PageLink to='#'>
              <span>&raquo;</span>
            </PageLink>
          </PageItem>
        </Pagination>
      </nav>
    </nav>
  </>
);
