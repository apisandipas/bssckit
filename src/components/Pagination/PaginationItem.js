import styled, { css } from 'styled-components';
import defaultTheme from '../../theme';
import { getColor, getBorder } from '../../utils/themeFunctions';
import { Li } from '../Utilities/base';
import PaginationLink from './PaginationLink';

const PaginationItem = styled(Li)`
  &:not(:first-child) {
    margin-left: -${props => getBorder(props, 'paginationLink', 'default')};
  }

  ${props =>
    props.active &&
    css`
      & ${PaginationLink} {
        z-index: 3;
        color: ${getColor(props, 'paginationLink', 'color')};
      }
    `}

  ${props =>
    props.disabled &&
    css`
      & ${PaginationLink} {
        color: ${getColor(props, 'paginationLink', 'disabledColor')};
        pointer-events: none;
        background-color: ${getColor(props, 'paginationLink', 'disabledBackground')};
        border-color: ${getColor(props, 'paginationLink', 'disabledBorderColor')};
      }
    `}
`;

PaginationItem.defaultProps = {
  theme: defaultTheme,
};

export default PaginationItem;
