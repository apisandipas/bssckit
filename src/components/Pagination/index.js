export { default as Pagination } from './Pagination';
export { default as PageLink } from './PageLink';
export { default as PageItem } from './PageItem';
