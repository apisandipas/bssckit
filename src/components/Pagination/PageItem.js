import styled, { css } from 'styled-components';
import defaultTheme from '../../theme';
import { getColor, getBorder, getMargin } from '../../utils/themeFunctions';
import { Li } from '../Utilities/base';
import PageLink from './PageLink';

const PageItem = styled(Li)`
  &:not(:first-child) {
    margin-left: -${props => getMargin(props, 'paginationLink', 'default')};
  }

  ${props =>
    props.active &&
    css`
      & ${PageLink} {
        z-index: 3;
        color: ${getColor(props, 'paginationLink', 'activeColor')} !important;
        background: ${getColor(props, 'paginationLink', 'activeBackgroundColor')};
        border-color: ${getColor(props, 'paginationLink', 'activeBorderColor')};
      }
    `}

  ${props =>
    props.disabled &&
    css`
      & ${PageLink} {
        color: ${getColor(props, 'paginationLink', 'disabledColor')} !important;
        pointer-events: none;
        background-color: ${getColor(props, 'paginationLink', 'disabledBackground')};
        border-color: ${getColor(props, 'paginationLink', 'disabledBorderColor')};
      }
    `}
`;

PageItem.defaultProps = {
  theme: defaultTheme,
};

export default PageItem;
