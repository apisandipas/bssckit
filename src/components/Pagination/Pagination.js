import styled, { css } from 'styled-components';
import defaultTheme from '../../theme';
import { getPadding, getBorderRadius } from '../../utils/themeFunctions';
import { Ul } from '../Utilities/base';
import PageItem from './PageItem';
import PageLink from './PageLink';

const padding = props => {
  if (props.lg) {
    return css`
      ${PageItem} {
      }

      ${PageLink} {
        padding: ${getPadding(props, 'paginationLink', 'lg')};
      }
    `;
  } else if (props.sm) {
    return css`
      ${PageItem} {
      }

      ${PageLink} {
        padding: ${getPadding(props, 'paginationLink', 'sm')};
      }
    `;
  } else {
    return css`
      ${PageItem} {
      }

      ${PageLink} {
        padding: ${getPadding(props, 'paginationLink', 'default')};
      }
    `;
  }
};

const borderRadius = props => {
  if (props.lg) {
    return css`
      ${PageItem}:first-child {
        a {
          border-top-left-radius-left: ${getBorderRadius(props, 'paginationLink', 'lg')};
          border-bottom-left-radius: ${getBorderRadius(props, 'paginationLink', 'lg')};
        }
      }

      ${PageItem}:last-child {
        a {
          border-top-right-radius: ${getBorderRadius(props, 'paginationLink', 'lg')};
          border-bottom-right-radius: ${getBorderRadius(props, 'paginationLink', 'lg')};
        }
      }
    `;
  } else if (props.sm) {
    return css`
      ${PageItem}:first-child {
        a {
          border-top-left-radius: ${getBorderRadius(props, 'paginationLink', 'sm')};
          border-bottom-left-radius: ${getBorderRadius(props, 'paginationLink', 'sm')};
        }
      }

      ${PageItem}:last-child {
        a {
          border-top-right-radius: ${getBorderRadius(props, 'paginationLink', 'sm')};
          border-bottom-right-radius: ${getBorderRadius(props, 'paginationLink', 'sm')};
        }
      }
    `;
  } else {
    return css`
      ${PageItem}:first-child {
        a {
          border-top-left-radius: ${getBorderRadius(props, 'paginationLink', 'default')};
          border-bottom-left-radius: ${getBorderRadius(props, 'paginationLink', 'default')};
        }
      }

      ${PageItem}:last-child {
        a {
          border-top-right-radius: ${getBorderRadius(props, 'paginationLink', 'default')};
          border-bottom-right-radius: ${getBorderRadius(props, 'paginationLink', 'default')};
        }
      }
    `;
  }
};

const Pagination = styled(Ul)`
  display: flex;
  margin-left: 0;
  list-style: none;

  ${props => padding(props)}
  ${props => borderRadius(props)}
  }
`;

Pagination.defaultProps = {
  theme: defaultTheme,
};

export default Pagination;
