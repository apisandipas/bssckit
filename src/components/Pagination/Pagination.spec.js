import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import Pagination from './Pagination';
import PageItem from './PageItem';
import PageLink from './PageLink';

describe('Pagination component', () => {
  it('Matches Pagination styles', () => {
    const { container } = render(<Pagination />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
    expect(container.firstChild).toHaveStyleRule('margin-left', '0');
    expect(container.firstChild).toHaveStyleRule('list-style', 'none');
  });
});

describe('PageItem component', () => {
  it('matches PageItem styles', () => {
    const { container } = render(<PageItem />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('PageLink component', () => {
  it('matches PageLink styles', () => {
    const { container } = render(<PageLink />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
