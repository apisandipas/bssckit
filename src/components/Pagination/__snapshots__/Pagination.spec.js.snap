// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageItem component matches PageItem styles 1`] = `
.c0:not(:first-child) {
  margin-left: -;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<li
  class="sc-AxhCb c0"
/>
`;

exports[`PageLink component matches PageLink styles 1`] = `
.c0 {
  position: relative;
  display: block;
  color: #007bff !important;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: 1px solid #dee2e6;
}

.c0:hover {
  z-index: 2;
  color: #005cbf;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: #e9ecef;
}

.c0:focus {
  z-index: 3;
  outline: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="c0"
/>
`;

exports[`Pagination component Matches Pagination styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0;
  list-style: none;
}

.c0 a {
  padding: 0.375rem 0.75rem;
}

.c0 .sc-fznWqX:first-child a {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.c0 .sc-fznWqX:last-child a {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<ul
  class="sc-AxiKw c0"
/>
`;
